cumulusci/cli/cci.py

```diff
@@ -259,42 +251,94 @@ def version():
     click.echo()
 
 
-@main.command(name="shell", help="Drop into a Python shell")
-def shell():
-    try:
-        config = load_config(load_project_config=True, load_keychain=True)
-    except SystemExit:
-        config = load_config(load_project_config=False, load_keychain=False)
+@cli.command(name="shell", help="Drop into a Python shell")
+@pass_runtime(require_project=False)
+def shell(runtime):
+    # alias for backwards-compatibility
+    config = runtime
 
     code.interact(local=dict(globals(), **locals()))
 
 
+GIST_404_ERR_MSG = """A 404 error code was returned when trying to create your gist.
+Please ensure that your GitHub personal access token has the 'Create gists' scope."""
+CCI_LOG_NOT_FOUND_MSG = """No logfile to open at path: {log_path}
+Please ensure you're running this command from the same directory you were experiencing an issue."""
+LAST_CMD_HEADER = "\n\n\nLast Command Run\n================================\n"
+
+
+@cli.command(name="gist", help="Create a gist from the latest logfile")
+@pass_runtime(require_project=False)
+def gist(runtime):
+    repo_root = RUNTIME.project_config.repo_root
+    log_path = f"{repo_root}/.cci/cci.log" if repo_root else f"cci.log"
+    try:
+        last_cmd_log = open(log_path, "r")
+    except FileNotFoundError:
+        click.echo(CCI_LOG_NOT_FOUND_MSG)
+        sys.exit(1)
```

Comment:
Raise a CumulusCIException. That leaves the decisions about how to report the error and how to exit in the hands of the `main` wrapper so it's consistent with other commands. For example that way it's possible to run `cci gist` with the `--debug` flag to see the traceback.